### PR TITLE
added category list extension

### PR DESCRIPTION
**Pull request by halyna-k**

### CODE REVIEW

**Review by djek195 — APPROVED** [review]

File: src/components/CategoryList/CategoryList.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import CategoryButton from '../../components/CategoryButton/CategoryButton';
+
+const CategoryList = (props) => {
+  const {
+    rowSpacing,
+    columnSpacing,
+    xsColumns,
+    smColumns,
+    mdColumns,
+    xlColumns,
```
Comment (dimalian) [inline src/components/CategoryList/CategoryList.js:12]: The component should be responsible for the correct display on different devices. You must choose the size of the columns and the spacing within the component.

File: src/components/CategoryList/CategoryList.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    array,
```
Comment (dimalian) [inline src/components/CategoryList/CategoryList.js:13]: rename to categories

File: src/components/CategoryList/CategoryList.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    onClick,
+  } = props;
+
+  return (
+    <Grid container rowSpacing={rowSpacing} columnSpacing={columnSpacing}>
+      {Array.from(array).map((el, index) => (
```
Comment (dimalian) [inline src/components/CategoryList/CategoryList.js:19]: Why not just write
```suggestion
      {array.map((el, index) => (
```

File: src/components/CategoryList/CategoryList.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        <Grid key={index} xs={xsColumns} sm={smColumns} md={mdColumns} xl={xlColumns}>
```
Comment (dimalian) [inline src/components/CategoryList/CategoryList.js:20]: If you have a unique value in the array, put it into the key parameter and avoid using the index. 
```suggestion
        <Grid key={el.name} xs={xsColumns} sm={smColumns} md={mdColumns} xl={xlColumns}>
```

File: src/pages/CatalogPage/CatalogPage.js [hunk 5]
```diff
@@ -1,93 +1,71 @@
...
-    image: 'http://placekitten.com/200/300',
-    alt: "Product 5",
-    price: 1200,
-  },
-];
+import { products, categories } from '../../data'
+
 
 const CatalogPage = () => {
+  const sm = useMediaQuery('(min-width: 480px)');
```
Comment (dimalian) [inline src/pages/CatalogPage/CatalogPage.js:17]: This option is more readable and has less chance to make a mistake.
```suggestion
  const sm = useMediaQuery(theme.breakpoints.up('sm'));
```

File: src/pages/CatalogPage/CatalogPage.js [hunk 6]
```diff
@@ -1,93 +1,71 @@
...
+          <CategoryList
+            array={categories.slice(0, showMore ? categories.length : ( md ? 4 : (sm ? 3 : 2)))}
+            rowSpacing={{ xs: 2.5 }}
+            columnSpacing={{ xs: 1 }}
+            xsColumns={6}
+            smColumns={4}
+            mdColumns={3}
+            onClick={handleClickCategoryButton} />
+          <Grid sx={{ marginX: 'auto', marginTop: 2.5 }}>
+            <CategoryMoreButton text={buttonText} onClick={handleClickShowMore} />
```
Comment (dimalian) [inline src/pages/CatalogPage/CatalogPage.js:55]:
```suggestion
            <CategoryMoreButton text={showMore ? 'Більше категорій' : 'Зменшити'} onClick={handleClickShowMore} />
```

File: src/pages/CatalogPage/CatalogPage.js [hunk 7]
```diff
@@ -1,93 +1,71 @@
...
+      ) : (
+        <Grid container sx={{ flexDirection: 'column', marginY: 2.5 }}>
+          <CategoryList
+            array={categories.slice(0, showMore ? categories.length : ( md ? 4 : (sm ? 3 : 2)))}
+            rowSpacing={{ xs: 2.5 }}
+            columnSpacing={{ xs: 1 }}
+            xsColumns={6}
+            smColumns={4}
+            mdColumns={3}
+            onClick={handleClickCategoryButton} />
```
Comment (dimalian) [inline src/pages/CatalogPage/CatalogPage.js:53]: All these conditions must be inside the CategoryList component. The parent component mustn't be worried about how to show the component on the different screens.

**Review by dimalian — CHANGES_REQUESTED** [review]

File: src/components/Advertising/Advertising.js [hunk 8]
```diff
@@ -8,14 +8,13 @@ const Advertising = () => {
...
-  useEffect(
-    () => changeImageIndex(),
-    []
-  );
+  useEffect(() => {
+    const changeImageIndex = () => {
+      setCurrentImage(randomNumber);
+    };
+    changeImageIndex();
+  }, [randomNumber]);
```
Comment (dimalian) [inline src/components/Advertising/Advertising.js]:
```suggestion
  useEffect(() => {
    const randomNumber = (Math.floor(Math.random() * 
  images.length));
      setCurrentImage(randomNumber);
  }, []);
```

File: src/components/CategoryList/CategoryList.js (hunk 2 above)
Comment (dimalian) [reply src/components/CategoryList/CategoryList.js]: this comment wasn't resolved

File: src/components/CategoryList/CategoryListMobile.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+import CategoryList from './CategoryList';
+import CategoryMoreButton from '../CategoryMoreButton/CategoryMoreButton'
+
+const CategoryListMobile = (props) => {
+  const { array, onClick } = props;
+  const theme = useTheme();
+  const sm = useMediaQuery(theme.breakpoints.up('sm'));
+  const md = useMediaQuery(theme.breakpoints.up('md'));
+  const [showMore, setShowMore] = useState(false);
+  const [buttonText, setButtonText] = useState('Більше категорій');
```
Comment (dimalian) [inline src/components/CategoryList/CategoryListMobile.js]: Don't create unnecessary useState. Show specific text according to the showMore variable.

File: src/components/CategoryList/CategoryListMobile.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        )}
+        rowSpacing={{ xs: 2.5 }}
+        columnSpacing={{ xs: 1 }}
+        xs={6}
+        sm={4}
+        md={3}
+        onClick={onClick}
+      />
+      <Grid sx={{ marginX: 'auto', marginTop: 2.5 }}>
+        <CategoryMoreButton text={buttonText} onClick={handleClickShowMore} />
```
Comment (dimalian) [inline src/components/CategoryList/CategoryListMobile.js]:
```suggestion
        <CategoryMoreButton text={showMore ? 'more' : 'less'} onClick={handleClickShowMore} />
```

**Review by dimalian — CHANGES_REQUESTED** [review]

**Review by dimalian — APPROVED** [review]

**Pull request merged**